src/assets/wise5/authoringTool/node/node-authoring/node-authoring.component.ts

```diff
@@ -163,35 +137,37 @@ class NodeAuthoringController {
   }
 
   private createPreviewURL(nodeId: string, constraints: boolean): string {
-    let previewURL = `${this.ConfigService.getConfigParam('previewProjectURL')}/${nodeId}`;
+    let previewURL = `${this.configService.getConfigParam('previewProjectURL')}/${nodeId}`;
     if (!constraints) {
       previewURL += '?constraints=false';
     }
     return previewURL;
   }
 
-  close() {
-    this.TeacherDataService.setCurrentNode(null);
+  close(): void {
+    this.teacherDataService.setCurrentNode(null);
     this.scrollToTopOfPage();
   }
 
-  showSaveErrorAdvancedAuthoring() {
-    alert(this.$translate('saveErrorAdvancedAuthoring'));
+  showSaveErrorAdvancedAuthoring(): void {
+    alert(
+      $localize`Error saving advanced authoring, possibly due to malformed JSON content. Check that your JSON is valid. Your changes have not been saved.`
+    );
   }
 
-  addComponent() {
-    this.$state.go('root.at.project.node.add-component.choose-component');
+  addComponent(): void {
```

Comment:
Add ```protected``` to all applicable functions in this class (functions that are only called from the template or this class).
```suggestion
  protected addComponent(): void {
```